addons/account/models/product.py

```diff
@@ -118,25 +118,26 @@ def _check_uom_not_in_invoice(self):
                 "If you want to change its Unit of Measure, please archive this product and create a new one."
             ))
 
+    def _force_default_tax(self, companies):
+        default_customer_taxes = companies.filtered('account_sale_tax_id').account_sale_tax_id
+        default_supplier_taxes = companies.filtered('account_purchase_tax_id').account_purchase_tax_id
+
+        products_without_company = self.filtered(lambda p: not p.company_id).sudo()
+        products_without_company.taxes_id += default_customer_taxes
+        products_without_company.supplier_taxes_id += default_supplier_taxes
+
+        products_without_company.invalidate_recordset(['taxes_id', 'supplier_taxes_id'])
+        self.invalidate_recordset(['taxes_id', 'supplier_taxes_id'])
```

Comment:
Why call on both? One is the subset of the other...
And nothing changed for the other...

Reply:
omg... say that to the original author of that line, or the one that approved his PR ;-) 

It's fine, it's fine, I'll do it -_-

Reply:
It wasn't cleaned it the past, it doesn't mean that we can't clean it anymore.
Since you are touching both lines, it is the perfect opportunity

Reply:
I know, I just pushed that change ;) . 

Also, reviewing that code made me realize why you might have bugged on the search not filtering products without company right away. It just doesn't belong there at all for me: I moved it to `create()`